lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsFileSystemConfig.java

```diff
@@ -30,13 +31,26 @@
 import java.util.concurrent.TimeUnit;
 
 import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.Locale.ENGLISH;
 
 public class GcsFileSystemConfig
 {
     public enum AuthType
     {
         ACCESS_TOKEN,
-        DEFAULT,
+        DEFAULT;
+
+        // Ensures backward compatibility with gcs.use-access-token config
+        // TRUE is mapped to ACCESS_TOKEN
+        // FALSE is mapped to DEFAULT
+        public static AuthType fromString(String value)
+        {
+            return switch (value.toUpperCase(ENGLISH)) {
```

Comment:
hacky. I also wanted backward compatibility here but I think deprecating `setUseAccessToken` but still keeping it. And add validation that `setUseAccessToken` is not used together with `setAuthType` would be cleaner